Another batch of spec fixes

kirkkwang opened the pull request:
This PR will fix another set of specs from the Rails 6.1/Hyrax 4/Blacklight 7 upgrade.  This will bring the failing spec count down from 26 to 10.

ref: https://github.com/scientist-softserv/ams/issues/67

<details><summary>Remaining Failures</summary>

```rb
Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Hyrax::EssenceTrackPresenter has tests
     # Add your tests here
     # ./spec/presenters/hyrax/essence_track_presenter_spec.rb:6

  2) FileSet has tests
     # Add your tests here
     # ./spec/models/file_set_spec.rb:4

  3) Bulkrax::CsvEntry exporting by Asset worktype writes the csv file with the asset and its children in a single row
     # Temporarily skipped with xit
     # ./spec/models/bulkrax/csv_entry_spec.rb:195

  4) Qa::LocalAuthority add some examples to (or delete) /app/samvera/hyrax-webapp/spec/models/qa/local_authority_spec.rb
     # Not yet implemented
     # ./spec/models/qa/local_authority_spec.rb:4

  5) Hyrax::EssenceTracksController has tests
     # Add your tests here
     # ./spec/controllers/hyrax/essence_tracks_controller_spec.rb:6

  6) Hyrax::AssetForm has tests
     # Add your tests here
     # ./spec/forms/hyrax/asset_form_spec.rb:6

  7) Hyrax::AssetsController has tests
     # Add your tests here
     # ./spec/controllers/hyrax/assets_controller_spec.rb:6

  8) Hyrax::Actors::PhysicalInstantiationActor has tests
     # Add your tests here
     # ./spec/actors/hyrax/actors/physical_instantiation_actor_spec.rb:6

  9) Search by dates Range * TO YYYY-MM returns all assets with a date on or before the last day of the specified year and month
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:110

  10) Search by dates Range YYYY-MM TO YYYY returns all assets with a date on or after the first day of the earlier year/month, and on or before the last day of the later year
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:154

  11) Search by dates Range * TO YYYY-MM-DD returns all assets with a date on or before the last day of the specified year and month
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:118

  12) Search by dates Range YYYY-MM TO YYYY-MM returns all assets with a date on or after the first day of the earlier year/month, and on or before the last day of the later year/month
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:167

  13) Search by dates Range * TO YYYY returns all assets with a date on or before the last day of the specified year
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:101

  14) Search by dates Range YYYY TO YYYY-MM-DD returns all assets with a date on or after the first day of the earlier year, and on or before the last year/month/day
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:145

  15) Search by dates Range YYYY TO YYYY returns all assets with a date on or after the first day of the earlier year, and on or before the last day of the later year
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:125

  16) Search by dates Range YYYY-MM-DD TO YYYY-MM returns all assets with a date on or after the first year/month/day and on or before the last day of the later year/month
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:203

  17) Search by dates Range YYYY-MM-DD TO YYYY-MM-DD returns all assets with a date on or after the first year/month/day and on or before the later year/month/day
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:214

  18) Search by dates Range YYYY-MM-DD TO YYYY returns all assets with a date on or after the first year/month/day and on or before the last day of the later year
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:191

  19) Search by dates with exact YYYY returns all assets with at least one date within that year
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:53

  20) Search by dates Range YYYY-MM TO YYYY-MM-DD returns all assets with a date on or after the first day of the earlier year/month, and on or before the later year/month/day
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:180

  21) Search by dates Exact YYYY-MM-DD returns all assets with a date on that exact day
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:70

  22) Search by dates Range YYYY TO * returns all assets with a date after the first day of the specified year
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:77

  23) Search by dates with exact YYYY-MM returns all assets with at least one date for that year and month
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:62

  24) Search by dates Range YYYY-MM TO * returns all assets with a date after the first of the specified year and month
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:86

  25) Search by dates Range YYYY TO YYYY-MM returns all assets with a date on or after the first day of the earlier year, and on or before the last day of the later year/month
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:135

  26) Search by dates Range YYYY-MM-DD TO * returns all assets with a date after the first of the specified year and month
     # No reason given
     # ./spec/features/search_by_dates_spec.rb:94

  27) Hyrax::DigitalInstantiationForm has tests
     # Add your tests here
     # ./spec/forms/hyrax/digital_instantiation_form_spec.rb:6

  28) Hyrax::DigitalInstantiationsController has tests
     # Add your tests here
     # ./spec/controllers/hyrax/digital_instantiations_controller_spec.rb:6

  29) Hyrax::Actors::EssenceTrackActor has tests
     # Add your tests here
     # ./spec/actors/hyrax/actors/essence_track_actor_spec.rb:6

  30) Hyrax::PhysicalInstantiationsController has tests
     # Add your tests here
     # ./spec/controllers/hyrax/physical_instantiations_controller_spec.rb:6

  31) Hyrax::DigitalInstantiationPresenter has tests
     # Add your tests here
     # ./spec/presenters/hyrax/digital_instantiation_presenter_spec.rb:6

  32) Qa::LocalAuthorityEntry add some examples to (or delete) /app/samvera/hyrax-webapp/spec/models/qa/local_authority_entry_spec.rb
     # Not yet implemented
     # ./spec/models/qa/local_authority_entry_spec.rb:4

  33) Hyrax::PhysicalInstantiationPresenter has tests
     # Add your tests here
     # ./spec/presenters/hyrax/physical_instantiation_presenter_spec.rb:6

  34) Hyrax::EssenceTrackForm has tests
     # Add your tests here
     # ./spec/forms/hyrax/essence_track_form_spec.rb:6

  35) Hyrax::Actors::DigitalInstantiationActor has tests
     # Add your tests here
     # ./spec/actors/hyrax/actors/digital_instantiation_actor_spec.rb:6

  36) Collection has tests
     # Add your tests here
     # ./spec/models/collection_spec.rb:4

  37) Hyrax::PhysicalInstantiationForm has tests
     # Add your tests here
     # ./spec/forms/hyrax/physical_instantiation_form_spec.rb:6

  38) InstantiationAdminData add some examples to (or delete) /app/samvera/hyrax-webapp/spec/models/instantiation_admin_data_spec.rb
     # Not yet implemented
     # ./spec/models/instantiation_admin_data_spec.rb:4


Failures:

  1) Pushes features #pushes gets the correct record set when navigating from a catalog search
     Failure/Error: click_link('Push To AAPB', class: 'aapb-push-button')

     Capybara::ElementNotFound:
       Unable to find link "Push To AAPB" with classes [aapb-push-button]
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:312:in `block in synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/base.rb:84:in `synchronize'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:301:in `synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:60:in `find'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/actions.rb:42:in `click_link'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/session.rb:773:in `click_link'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `call'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `click_link'
     # ./spec/features/pushes_spec.rb:48:in `block (3 levels) in <top (required)>'

  2) Create Asset with Asset Type Create adminset, create asset Create Asset with Asset Type
     Failure/Error: find('button.multiselect').click

     Capybara::ElementNotFound:
       Unable to find css "button.multiselect" within #<Capybara::Node::Element tag="div" path="/HTML/BODY[1]/DIV[2]/DIV[2]/FORM[1]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/DIV[2]/DIV[2]/DIV[2]/DIV[1]/DIV[1]">
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:312:in `block in synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/base.rb:84:in `synchronize'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:301:in `synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:60:in `find'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/session.rb:773:in `find'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `call'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `find'
     # ./spec/features/create_asset_with_genre_vocabulary_spec.rb:60:in `block (4 levels) in <top (required)>'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/session.rb:365:in `within'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `call'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `within_element'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/rspec/matcher_proxies.rb:15:in `within'
     # ./spec/features/create_asset_with_genre_vocabulary_spec.rb:59:in `block (3 levels) in <top (required)>'

  3) Create and Validate Asset Create adminset, create asset Create and Validate Asset, Search asset
     Failure/Error: expect(page).to have_content asset_attributes[:spatial_coverage]
       expected to find text "My Test Spatial coverage" in "AMS\nAMS\nSwitch language\nEnglish\nView\n user34@example.com\nprofile\nHome\nAbout\nHelp\nContact\nSearch in\nAll Fields\nTitles (All)\nSeries Title\nDescriptions (All)\nContributor\nCreator\nTitle\nAbstract or Summary\nPublisher\nDate Created\nSubject\nLanguage\nResource Type\nInstantiation Format\nLocal Id\nDepositor\nRights Statement\nLicense\nBroadcast\nCreated\nDate\nCopyright Date\nSpatial Coverage\nTemporal Coverage\nAudience Level\nAudience Rating\nRights Summary\nPbs Nola Code\nEidr\nLocal Instantiation Identifier\nPortrayal\nsearch for\nGo\nAll\nAll\nMore options\nHomeDashboardWorksTest Series Title 2; Test...\nAsset\n  Test Series Title 2; Test Series Title 11; Test Program Title 3; Test Program Title 12; Test Episode Number Title 14; Test Episode Number Title 5; Test Episode Title 13; Test Episode Title 4; Test Segment Title 15; Test Segment Title 6; Test Clip Title 9; Test Clip Title 18; Test Promo Title 17; Test Promo Title 8; Test Raw Footage Title 16; Test Raw Footage Title 7; Test Title 10; Test Title 1 Public Deposited\nEdit\nAttach Child\n\n\n\n\nNot playable: unrecognized media type: \"\".\nAsset Details\nCredits\nInstantiations\nThis Asset has no files associated with it. Click \"edit\" to add more files.\nIn Administrative Set:\nDefault Admin Set\nA service of Samvera.\nAMS v4.0.0\nCopyright © 2018 Samvera Licensed under the Apache License, Version 2.0". (However, it was found 1 time including non-visible text.)
     # ./spec/features/create_asset_spec.rb:148:in `block (3 levels) in <top (required)>'

  4) AdminAddUserroleAsAdminsetManager. As Admin add a UserRole as Adminset Manager Assign set of user (role) as Manager to AdminSet
     Failure/Error: expect(page).to have_content admin_set.title[0]
       expected to find text "Title 3" in "AMS\nAMS\nSwitch language\nEnglish\nView\n admin17@example.com\nprofile\nadmin17@example.com\nDashboard\nACTIVITY\nYour activity\nStatistics\nREPOSITORY CONTENTS\nCollections\nWorks\nImporters\nExporters\nTASKS\nReview Submissions\nManage Users\nManage Embargoes\nManage Leases\nCONFIGURATION\nSettings\nManage Users\nManage Role\nWorkflow Roles\nHomeDashboardCollections\nCollections\nAll Collections\nYour Collections\nAdd New Collection\n0 collections you own in the repository\nCollections listing\nFilter collections:\nSearch AMS\nGo\nListing of items you have deposited in AMS\nSelect all files to be added to a collection or edited\nSelect\nSelect to access selection option\nTitle Type Last modified Items Visibility of Collection Actions"
     # ./spec/features/admin_add_userrole_as_adminset_manager_spec.rb:19:in `block (3 levels) in <top (required)>'

  5) Create and Validate Asset,Digital Instantiation, EssenseTrack Create adminset, create asset, import pbcore xml for digital instantiation and essensetrack Create and Validate Asset, Search asset
     Failure/Error: find('button.multiselect').click

     Capybara::ElementNotFound:
       Unable to find css "button.multiselect" within #<Capybara::Node::Element tag="div" path="/HTML/BODY[1]/DIV[2]/DIV[2]/FORM[1]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/DIV[2]/DIV[3]/DIV[2]/DIV[1]/DIV[2]">
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:312:in `block in synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/base.rb:84:in `synchronize'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:301:in `synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:60:in `find'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/session.rb:773:in `find'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `call'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `find'
     # ./spec/features/create_asset_with_digitalinstantiation_spec.rb:137:in `block (4 levels) in <top (required)>'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/session.rb:365:in `within'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `call'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `within_element'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/rspec/matcher_proxies.rb:15:in `within'
     # ./spec/features/create_asset_with_digitalinstantiation_spec.rb:136:in `block (3 levels) in <top (required)>'

  6) Create Asset with Asset Type Create adminset, create asset Create Asset with Asset Type
     Failure/Error: find('button.multiselect').click

     Capybara::ElementNotFound:
       Unable to find css "button.multiselect" within #<Capybara::Node::Element tag="div" path="/HTML/BODY[1]/DIV[2]/DIV[2]/FORM[1]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/DIV[1]/DIV[2]/DIV[1]/DIV[2]/DIV[1]/DIV[6]">
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:312:in `block in synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/base.rb:84:in `synchronize'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:301:in `synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:60:in `find'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/session.rb:773:in `find'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `call'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `find'
     # ./spec/features/create_asset_with_asset_types_vocabulary_spec.rb:58:in `block (4 levels) in <top (required)>'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/session.rb:365:in `within'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `call'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `within_element'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/rspec/matcher_proxies.rb:15:in `within'
     # ./spec/features/create_asset_with_asset_types_vocabulary_spec.rb:57:in `block (3 levels) in <top (required)>'

  7) AssignMultipleRolesAsManager. Add Manager permissions to user (Role) Assign set of user (role) as Manager to multiple AdminSet
     Failure/Error: expect(page).to have_content 'You are not authorized to access this page.'
       expected to find text "You are not authorized to access this page." in "AMS\nAMS\nSwitch language\nEnglish\nView\n user252@example.com\nprofile\nuser252@example.com\nDashboard\nACTIVITY\nYour activity\nREPOSITORY CONTENTS\nCollections\nWorks\nImporters\nExporters\nHomeDashboardCollections\nCollections\n0 collections you own in the repository\nCollections listing\nFilter collections:\nSearch AMS\nGo\nListing of items you have deposited in AMS\nSelect all files to be added to a collection or edited\nSelect\nSelect to access selection option\nTitle Type Last modified Items Visibility of Collection Actions"
     # ./spec/features/admin/admin_sets/add_multiple_userrole_as_manager_spec.rb:62:in `block (3 levels) in <top (required)>'

  8) PushesController GET /pushes/index assigns @pushes to all Push model instances
     Failure/Error: before { get :index }

     NoMethodError:
       undefined method `formats' for #<ActionView::Template app/views/pushes/index.html.erb locals=[]>
       Did you mean?  format
     # /usr/local/bundle/gems/react-rails-2.7.1/lib/react/rails/controller_lifecycle.rb:31:in `use_react_component_helper'
     # /usr/local/bundle/gems/rails-controller-testing-1.0.5/lib/rails/controller/testing/template_assertions.rb:62:in `process'
     # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:35:in `block in process'
     # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:104:in `catch'
     # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:104:in `_catch_warden'
     # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:35:in `process'
     # /usr/local/bundle/gems/rails-controller-testing-1.0.5/lib/rails/controller/testing/integration.rb:16:in `block (2 levels) in <module:Integration>'
     # ./spec/controllers/pushes_controller_spec.rb:16:in `block (3 levels) in <top (required)>'

  9) PushesController GET /pushes/:id assign @push to the Push instance for the ID given
     Failure/Error: before { get :show, params: { id: push.id } }

     NoMethodError:
       undefined method `formats' for #<ActionView::Template app/views/pushes/show.html.erb locals=[]>
       Did you mean?  format
     # /usr/local/bundle/gems/react-rails-2.7.1/lib/react/rails/controller_lifecycle.rb:31:in `use_react_component_helper'
     # /usr/local/bundle/gems/rails-controller-testing-1.0.5/lib/rails/controller/testing/template_assertions.rb:62:in `process'
     # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:35:in `block in process'
     # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:104:in `catch'
     # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:104:in `_catch_warden'
     # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:35:in `process'
     # /usr/local/bundle/gems/rails-controller-testing-1.0.5/lib/rails/controller/testing/integration.rb:16:in `block (2 levels) in <module:Integration>'
     # ./spec/controllers/pushes_controller_spec.rb:24:in `block (3 levels) in <top (required)>'

  10) Admin::UsersController GET #new returns a success response
      Failure/Error: get :new

      NoMethodError:
        undefined method `formats' for #<ActionView::Template:0x0000ffff370b8830>
        Did you mean?  format
      # /usr/local/bundle/gems/react-rails-2.7.1/lib/react/rails/controller_lifecycle.rb:31:in `use_react_component_helper'
      # /usr/local/bundle/gems/rails-controller-testing-1.0.5/lib/rails/controller/testing/template_assertions.rb:62:in `process'
      # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:35:in `block in process'
      # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:104:in `catch'
      # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:104:in `_catch_warden'
      # /usr/local/bundle/gems/devise-4.9.2/lib/devise/test/controller_helpers.rb:35:in `process'
      # /usr/local/bundle/gems/rails-controller-testing-1.0.5/lib/rails/controller/testing/integration.rb:16:in `block (2 levels) in <module:Integration>'
      # ./spec/controllers/admin/users_controller_spec.rb:10:in `block (3 levels) in <top (required)>'

Deprecation Warnings:

Using `should` from rspec-expectations' old `:should` syntax without explicitly enabling the syntax is deprecated. Use the new `:expect` syntax or explicitly enable `:should` with `config.expect_with(:rspec) { |c| c.syntax = :should }` instead. Called from /app/samvera/hyrax-webapp/spec/services/aapb/batch_ingest/zipped_pbcore_reader_spec.rb:25:in `block (4 levels) in <top (required)>'.


If you need more of the backtrace for any of these deprecations to
identify where to make the necessary changes, you can configure
`config.raise_errors_for_deprecations!`, and it will turn the
deprecation warnings into errors, giving you the full backtrace.

1 deprecation warning total

Finished in 14 minutes 12 seconds (files took 9.07 seconds to load)
630 examples, 10 failures, 38 pending

Failed examples:

rspec ./spec/features/pushes_spec.rb:42 # Pushes features #pushes gets the correct record set when navigating from a catalog search
rspec ./spec/features/create_asset_with_genre_vocabulary_spec.rb:31 # Create Asset with Asset Type Create adminset, create asset Create Asset with Asset Type
rspec ./spec/features/create_asset_spec.rb:67 # Create and Validate Asset Create adminset, create asset Create and Validate Asset, Search asset
rspec ./spec/features/admin_add_userrole_as_adminset_manager_spec.rb:15 # AdminAddUserroleAsAdminsetManager. As Admin add a UserRole as Adminset Manager Assign set of user (role) as Manager to AdminSet
rspec ./spec/features/create_asset_with_digitalinstantiation_spec.rb:63 # Create and Validate Asset,Digital Instantiation, EssenseTrack Create adminset, create asset, import pbcore xml for digital instantiation and essensetrack Create and Validate Asset, Search asset
rspec ./spec/features/create_asset_with_asset_types_vocabulary_spec.rb:31 # Create Asset with Asset Type Create adminset, create asset Create Asset with Asset Type
rspec ./spec/features/admin/admin_sets/add_multiple_userrole_as_manager_spec.rb:15 # AssignMultipleRolesAsManager. Add Manager permissions to user (Role) Assign set of user (role) as Manager to multiple AdminSet
rspec ./spec/controllers/pushes_controller_spec.rb:17 # PushesController GET /pushes/index assigns @pushes to all Push model instances
rspec ./spec/controllers/pushes_controller_spec.rb:25 # PushesController GET /pushes/:id assign @push to the Push instance for the ID given
rspec ./spec/controllers/admin/users_controller_spec.rb:9 # Admin::UsersController GET #new returns a success response

Randomized with seed 30145
```

</details>

Commit:
✅ Use Blacklight's `Field` object

This commit will update the view to use Blacklight::Configuration::Field
object for Blacklight 7.0 compatibility.

Ref:
  - https://github.com/projectblacklight/blacklight/pull/2307

Commit:
✅ Use `Sipity.Agent` instead of `PowerConverter`

This commit updates the `#sipity_agent` method to use `Sipity.Agent` in
Hyrax 4.0 because `PowerConverter` is deprecated.  This also fixed the
previously skipped `aapb_pbcore_zipped_spec`.

Ref:
  - https://github.com/samvera/hyrax/pull/5980

Commit:
✅ Fix `polymorphic_path` method signature

In the specs we were getting:

```
ActionView::Template::Error:
  Please use symbols for polymorphic route arguments.
```

This commit will fix that for all `show_actions` partials.

Ref:
  - https://github.com/rails/rails/releases/tag/v6.0.3.7
  @see
    > Prevent string polymorphic route arguments.

Commit:
✅ Fix the way #query_solr for Hyrax 4.0

This commit will adjust how `#query_solr` works for the upgrade to Hyrax
4.0 which uses Blacklight 7.  The `#search_results` method no longer
takes an argument, so we need to adjust how we call it.  I added the
same method as from Hyrax to call the `Hyrax::SearchService`.  I
couldn't call it directly from Hyrax because (at least in the tests) I
was not able to pass the params because they were not permitted.

Ref
  - https://github.com/samvera/hyrax/blob/4.x-stable/app/controllers/concerns/hyrax/controller.rb#L48-L51

ShanaLMoore commented on app/views/catalog/_index_list_default.html.erb:
```diff
@@ -8,7 +8,7 @@
 <% doc_presenter = index_presenter(document) %>
 
-<% if(doc_presenter.field_value('has_model_ssim') == 'Collection') %>
+<% if field_value_for(doc_presenter, 'has_model_ssim') == 'Collection' %>
```
Should we add a conditional here? Is this method available in the production code?

kirkkwang replied:
From the specs, I observed that this doesn't break anything in the current Blacklight but I can wrap it just in case

ShanaLMoore commented on app/views/hyrax/assets/_show_actions.html.erb:
```diff
@@ -31,7 +31,7 @@
               <% presenter.valid_child_concerns.each do |concern| %>
                 <li>
-                  <%= link_to "Attach #{concern.human_readable_type}", polymorphic_path([main_app, :new, :hyrax, :parent, concern.model_name.singular], parent_id: presenter.id) %>
+                  <%= link_to "Attach #{concern.human_readable_type}", polymorphic_path([main_app, :new, :hyrax, :parent, concern.model_name.singular.to_sym], parent_id: presenter.id) %>
```
Should we add a conditional here (and in the other places w this change)? Will calling #to_sym break things in production code?

kirkkwang replied:
From the specs, I observed that this doesn't break anything in the current Rails but I can wrap it up just in case

ShanaLMoore commented on spec/features/batch_ingest/aapb_pbcore_zipped_spec.rb:
```diff
@@ -5,96 +5,95 @@
-      end
-    # TODO: Tests for invalid Assets.
-    else
-      skip 'Until we decide if we keep Hyrax::BatchIngest, see: https://github.com/scientist-softserv/ams/issues/58'
```
You removed the skip here. Was that intentional?

kirkkwang replied:
Yup, this [commit](https://github.com/WGBH-MLA/ams/pull/754/commits/5ac83eef15e4c5a84c7137b83b0b1b810094b197) fixed it

Commit:
🧹 Wrap recent changes in conditional

This commit wraps the `App.rails_5_1?` conditional around the recent
changes just in case it breaks anything for production app.  The tests
indicate it would not however we don't know how comprehensive the tests
are at this moment.  For `_index_list_default.html.erb`, I decided it
would be easier to just wrap the entire file in the conditional rather
than the specific parts.  This would make it easier to revert as well.
